sdks/python/apache_beam/ml/inference/base.py

```diff
@@ -339,6 +346,139 @@ def should_garbage_collect_on_timeout(self) -> bool:
     return self.share_model_across_processes()
 
 
+class RemoteModelHandler(ABC, ModelHandler[ExampleT, PredictionT, ModelT]):
+  """Has the ability to call a model at a remote endpoint."""
+  def __init__(
```

Comment:
just an idea to use config classes since the init arg list could be long (might be not):
```
from dataclasses import dataclass, field
from typing import Callable

@dataclass
class RemoteModelHandlerConfig:
    namespace: str = ''
    num_retries: int = 5
    throttle_delay_secs: int = 5
    retry_filter: Callable[[Exception], bool] = field(default_factory=lambda: lambda x: True)
    window_ms: int = 1 * _MILLISECOND_TO_SECOND
    bucket_ms: int = 1 * _MILLISECOND_TO_SECOND
    overload_ratio: float = 2
    ```

Reply:
maybe @shunping has some good ideas.

Reply:
I know this is largely coming from a perspective of how to surface this for yaml, but is this configuration consideration actually blocking here? We have to surface all of these parameters at some point

Reply:
I do not think it blocks your work. I just want to call out whether we want to do more like https://github.com/apache/beam/blob/master/sdks/python/apache_beam/ml/anomaly/transforms.py#L61. If the arguments keep growing, a config class might be better.

Reply:
I mainly ask because you didn't approve the PR

Reply:
Oh, https://github.com/apache/beam/pull/34379#issuecomment-2773264712